src/app/chip-zcl-zpro-codec.h

```diff
@@ -136,6 +136,204 @@ uint16_t encodeIdentifyQueryCommand(uint8_t * buffer, uint16_t buf_length, uint8
  * */
 uint16_t encodeIdentifyCommand(uint8_t * buffer, uint16_t buf_length, uint8_t destination_endpoint, uint16_t duration);
 
+/**
+ * Color control cluster commands
+ * */
+
+/**
+ * @brief Encode a move-to-hue command for the Color Control cluster
+ * @param buffer                Buffer to encode into
+ * @param buf_length            Length of buffer
+ * @param destination_endpoint  Destination endpoint
+ * @param hue                   The hue value of the color
+ * @param direction             A direction to change the hue to
+ * @param transitionTime        The number of steps for the color transition
+ * @param optionMask            An option mask
+ * @param optionOverride        An option override
+ * */
+uint16_t encodeMoveToHueCommand(uint8_t * buffer, uint16_t buf_length, uint8_t destination_endpoint, uint8_t hue, uint8_t direction,
```

Comment:
instead of a function per command, can we bind a printf-style string to each command and have a data-driven vararg encode() function?

Reply:
something like:

```
encodeCommand(buf, len, command_id, "%u8%u8%u8",  destination_endpoint, hue, direction, ...); 
```

Reply:
> something like:
> 
> ```
> encodeCommand(buf, len, command_id, "%u8%u8%u8",  destination_endpoint, hue, direction, ...); 
> ```

Do you want to discuss the approach in https://github.com/project-chip/connectedhomeip/pull/2978 ? :)